core/trino-main/src/main/java/io/trino/connector/ConnectorManager.java

```diff
@@ -410,10 +397,10 @@ private Connector createConnector(CatalogName catalogName, InternalConnectorFact
                 new InternalMetadataProvider(metadataManager, typeManager),
                 pageSorter,
                 pageIndexerFactory,
-                factory.getDuplicatePluginClassLoaderFactory(catalogName));
+                duplicatePluginClassLoaderFactory);
```

Comment:
Why do this when you already have the catalog name here?

Actually, this is broken since the `ConnectorContextInstance` constructor is not changed.

Reply:
We are passing in a CatalogClassLoaderSupplier for the class loader factory here.  This supplier tracks the creation of a duplicated classloader, and when that happens it triggers a classloader registration with the handle resolver.  The destroy method for the CatalogClassLoaderSupplier is called when the Materialized catalog is shutdown, and this method unregisters the classloader.